runtime/interpreter/encode.go

```diff
@@ -416,7 +416,9 @@ func (e *Encoder) prepare(
 		return e.prepareTypeValue(v)
 
 	default:
-		return nil, fmt.Errorf("unsupported value: %[1]T, %[1]v", v)
+		return nil, EncodingUnsupportedValueError{
```

Comment:
Should we add the path to the error as well?

Reply:
Good idea, doesn't hurt. Added in 9a02b9514b5f822db9560d461377de848f544d4e